src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts

```diff
@@ -14,25 +17,36 @@ import { Observable } from 'rxjs/Observable';
 export class DotEditLayoutComponent implements OnInit {
     @ViewChild('editLayoutGrid') editLayoutGrid: DotEditLayoutGridComponent;
 
-    pageView: Observable<DotPageView>;
+    grid: Observable<DotLayoutGridBox[]>;
+    page: DotPageView;
 
-    constructor(public router: Router, private route: ActivatedRoute, private pageViewService: PageViewService) {}
+    constructor(
+        private dotEditLayoutService: DotEditLayoutService,
+        private pageViewService: PageViewService,
+        private route: ActivatedRoute,
+        public router: Router
+    ) {}
 
     ngOnInit(): void {
-        this.pageView = this.route.data.pluck('pageView');
+        this.grid = this.route.data
+            .pluck('pageView')
+            .do((pageView: DotPageView) => {
+                this.page = pageView;
+            })
+            .map((pageView: DotPageView) => {
+                return pageView.layout.body.rows.length
+                    ? this.dotEditLayoutService.getDotLayoutGridBox(pageView)
+                    : DOT_LAYOUT_DEFAULT_GRID;
+            });
     }
 
-    public saveLayout() {
-        const layoutBody: DotLayoutBody = this.editLayoutGrid.getLayoutBody();
-        this.pageView.subscribe(pageView => {
-            this.pageViewService
-                .save(<DotPageView>{
-                    page: pageView.page,
-                    layout: {
-                        body: layoutBody
-                    }
-                })
-                .subscribe();
-        });
+    /**
+     * Get the LayoutBody and call the service to save the layout
+     *
+     */
+    saveLayout() {
+        const layoutBody: DotLayoutBody = this.dotEditLayoutService.getDotLayoutBody(this.editLayoutGrid.getGrid());
+        const pageView: DotPageView = Object.assign({}, this.page, { layout: { body: layoutBody } });
+        this.pageViewService.save(pageView).subscribe();
```

Comment:
When this is 100% done... we'll be probably sending a ```DotPageView``` to save, so I'm not 100% sure if the endpoint expect that, but I'm leaving this as close to finish as I can.